.github/workflows/test.yml

```
@@ -0,0 +1,60 @@
+name: Test
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
```

This means that this workflow runs if the pull request merges into `main` ([source](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-branches)).

I don't see a need to not include any target branch in this workflow.

```suggestion
  pull_request:
```

Same suggestion for all other *.yml files.